### DISCUSSION TITLE
LightWand

### DISCUSSION THREAD

**Discussion by digitaleJugend** [MartinNohr/MagicImageWand#64] [Q&A]
Hi Martin,

i starting to update my old lightwand painter from Michael Ross with the Arduino Mega and LCD Screen.
Wich timer.h library do you use(line 42)? I didn't have the right library. 

Thank you.

Christian

**Comment by MartinNohr** [comment]
I'll need a little more information, line 42 from which file?
Also, please note that this version does not use the Mega, it uses a TTGO T-Display which is based on the ESP32.
There is no timer.h file in any case. I do use some timers, but they are part of the esp32 RTOS system. Also note that this code will not run on a Mega at all, it will only run on ESP32.

**Comment by MartinNohr** [comment]
I think this is the version you want: https://github.com/MartinNohr/LightWand. Timer.h is on line 42 I see.
It was designed to run on the Mega. I don't remember what version of timer.h I used on that. I'll see if I can figure it out. I haven't worked on that code for several years now. The ESP32 TTGO is where I concentrate my efforts. It is a much faster chip with enough memory and WiFi. It even has a web server that allows a browser to change settings and run images.

**Comment by MartinNohr** [comment]
I think this is the timer code I used. Note that he has renamed the .h file since I used it last.
https://github.com/contrem/arduino-timer

**Comment by MartinNohr** [comment]
I posted a reply on the github site, I also added the github link to the correct code if you want to use the Mega. The code on this current site will only run on ESP32 now. The Mega was too slow and not enough memory for what I wanted to do.

The timer is: contrem/arduino-timer: Non-blocking library for delaying function calls (github.com) <https://github.com/contrem/arduino-timer>

**Reply by digitaleJugend** [reply]
Thanks Mark the code worked with the correct library( https://github.com/contrem/arduino-timer). You are right the mega is really lame. The Mega hangs sometimes and the graphics are displayed late. We often do workshops with children and they are very impatient when something does not work right away.
I still have a few ESP32 lying around and will try your TTGO version. Your PCB looks very nice. Would it be possible to order a PCB from you? I like the solution to control it also via a web-browser. Here in Germany I have found the ready solution of,but I don't know if this is better from the processor performance: https://www.ulrichradig.de/home/index.php/dmx/light-painter

Many greetings from Germany.
Christian

**Comment by MartinNohr** [comment]
I have two PCB's, one with and one without an onboard voltage regulator. The onboard voltage regulator is easier to build, with the other one you need an external 5V supply. I sell the PCB with the onboard voltage regulator with a programmed TTGO for $45 plus shipping. The one without is $40. Without the TTGO $30 and $25. The bad news is shipping to Germany is not cheap. Get me an address and I'll find out how much. 
Read the docx file to see all the features of the MIW. Note that it also has DMX control now.
The link you provided to the radig solution is interesting, I had never heard of that one. The TTGO runs at 240MHz and is a dual processor. It is nice and fast.
I used to speak pretty good german, but I haven't used it for many years. I can still read german though, I went to the link you sent and didn't need to translate it! I was born in Aarhus, Denmark, left at 18 when my parents immigrated to Canada.
If you send an email to mnohr247@gmail.com (my spam catcher email), I'll watch for it and then send you my real email address.

**Reply by MartinNohr** [reply]
I haven't seen any email from you, if I missed it please send again. Thanks.

**Comment by kajastudio** [comment]
Hi Martin. I have a problem with Lilygo TTGO T Display. I tried to program the board using Flash Download Tool. I downloaded the bin file from your GitHub, but the board does not start. The board I used I bought on Aliexpress.
https://www.aliexpress.com/item/33050639690.html?pdp_npi=4%40dis%21PLN%2151%2C42%20z%C5%82%2151%2C41%20z%C5%82%21%21%2113.15%2113.15%21%40211b61b017246677371887722e0def%2112000037296934408%21sh%21PL%21715956879%21X&spm=a2g0o.store_pc_allItems_or_groupList.new_all_items_2007599132036.33050639690

I made a test circuit according to the schematic without a power supply. I intend to power the MIW from a power bank, I do not need full light output to take pictures, I usually take pictures in very dark places and I do not think I will use more than 20% brightness. Tell me if the bin file is compatible with this board or maybe I am doing something wrong. With the same software I upload the firmware from the website: https://github.com/Xinyuan-LilyGO/TTGO-T-Display/tree/master/firmware
and everything works fine. I cannot compile the code myself, I am interested in taking pictures, I have enough technical skills to cope with board design, soldering etc. but compiling the code is a bit difficult for me. Thanks for any help. Best regards, Jarek
